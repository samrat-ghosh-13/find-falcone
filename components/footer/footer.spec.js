// react
import React from "react";

// component
import Footer from "./footer";

// jest renderer
import renderer from "react-test-renderer";

// testing library react
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

describe("Test Footer component", () => {
  it("Case 1: Footer Component Renders Correctly (Snapshot)", () => {
    const headerComponentTree = renderer.create(<Footer />).toJSON();
    expect(headerComponentTree).toMatchSnapshot();
  });
  it("Case 2: Checks the Footer text is correct or not", () => {
    const { queryByTestId } = render(<Footer />);
    expect(
      queryByTestId("app__footer__contents", { exact: true }).innerHTML
    ).toBe(
      'Made with <span class="app__footer__contents__logo">❤️</span> by<a href="https://www.linkedin.com/in/samratat/" target="_blank" rel="noreferrer">Samrat Ghosh</a>© 2021'
    );
  });
});
