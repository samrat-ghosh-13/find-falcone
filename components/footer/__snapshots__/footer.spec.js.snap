// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Footer component Case 1: Footer Component Renders Correctly (Snapshot) 1`] = `
<article>
  <footer
    className="app__footer"
    data-testid="app__footer"
  >
    <div
      className="app__footer__contents"
      data-testid="app__footer__contents"
    >
      Made with 
      <span
        className="app__footer__contents__logo"
      >
        ❤️
      </span>
       by
      <a
        href="https://www.linkedin.com/in/samratat/"
        rel="noreferrer"
        target="_blank"
      >
        Samrat Ghosh
      </a>
      © 2021
    </div>
  </footer>
</article>
`;
